api/src/shared/infrastructure/jobs/JobPgBoss.js

```diff
@@ -1,27 +1,44 @@
+import { DomainTransaction } from '../../domain/DomainTransaction.js';
+
 class JobPgBoss {
-  constructor(config, queryBuilder) {
+  constructor(config) {
     this.name = config.name;
     this.retryLimit = config.retryLimit ?? 0;
     this.retryDelay = config.retryDelay ?? 30;
     this.retryBackoff = config.retryBackoff || false;
     this.expireIn = config.expireIn || '00:15:00';
-    this.queryBuilder = queryBuilder;
   }
 
-  async schedule(data) {
-    await this.queryBuilder('pgboss.job').insert({
+  #buildPayload(data) {
+    return {
       name: this.name,
       retrylimit: this.retryLimit,
       retrydelay: this.retryDelay,
       retrybackoff: this.retryBackoff,
       expirein: this.expireIn,
       data,
       on_complete: true,
-    });
+    };
+  }
+
+  async #schedule(jobs) {
```

Comment:
Je propose de profiter du renommage pour supprimer cette confusion possible entre notre nommage qui pointe vers 'insert' de pgboss alors qu'il existe une methode `schedule` [cote pgboss](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#schedulename-cron-data-options)

```suggestion
  async #insert(jobs) {
```

Reply:
Note : je propose send pour se rapprocher d'un working des queues ou celui qui envoie est souvent nomme 'producer' ou 'sender'